File: leetcode/completed/2373.py
from typing import List

class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        n = len(grid)
        res = []

        for r in range(1, n-1):
            temp = []
            for c in range(1, n-1):
                t = 0
                for dr in range(-1, 2):
                    for dc in range(-1, 2):
                        t = max(t, grid[r+dr][c+dc])
                temp.append(t)
            res.append(temp)

        return res

File: leetcode/completed/2373.java
public class 2373 {
    class Solution {
        public int[][] largestLocal(int[][] grid) {
            int n = grid.length;
            int[][] res = new int[n-2][n-2];
    
            for (int r = 1; r < n-1; r++) {
                for (int c = 1; c < n-1; c++) {
                    int t = 0;
                    for (int dc = -1; dc < 2; dc++) {
                        for (int dr = -1; dr < 2; dr++) {
                            t = Math.max(t, grid[r+dr][c+dc]);
                        }
                    }
                    res[r-1][c-1] = t;
                }
            } 
    
            return res;
        }
    }
}
